tests/common/util.rs

```diff
@@ -1191,26 +1221,613 @@ impl UCommand {
+        match self.captured_stderr.as_mut() {
+            Some(output) => output.output_bytes(),
+            None if self.raw.stderr.is_some() => {
+                let mut buffer: Vec<u8> = vec![];
+                let stderr = self.raw.stderr.as_mut().unwrap();
+                stderr.read_to_end(&mut buffer).unwrap();
+                buffer
+            }
+            None => vec![],
+        }
+    }
+
+    /// Reads and returns all output from start of the child process until now. Each subsequent call
+    /// of any of the methods below will operate on the subsequent output of the child process. Note
+    /// that this function will work like [`UChild::stderr_bytes`] if the output wasn't captured
+    /// with [`CapturedOutput`], for example if [`UCommand::set_stderr`] was used with
+    /// [`Stdio::piped`].
+    ///
+    /// * [`UChild::stderr`]
+    /// * [`UChild::stderr_bytes`]
+    /// * [`UChild::stderr_exact_bytes`]
+    pub fn stderr_all_bytes(&mut self) -> Vec<u8> {
+        match self.captured_stderr.as_mut() {
+            Some(output) => output.output_all_bytes(),
+            None => self.stderr_bytes(),
+        }
+    }
+
+    /// Reads, consumes and returns the exact amount of bytes from stderr. If stderr is redirect to
+    /// stdout with [`UCommand::stderr_to_stdout`] then always zero bytes are returned.
+    ///
+    /// # Important
+    /// This method blocks indefinitely if the `size` amount of bytes cannot be read.
+    pub fn stderr_exact_bytes(&mut self, size: usize) -> Vec<u8> {
+        match self.captured_stderr.as_mut() {
+            Some(output) => output.output_exact_bytes(size),
+            None if self.raw.stderr.is_some() => {
+                let stderr = self.raw.stderr.as_mut().unwrap();
+                let mut buffer = vec![0; size];
+                stderr.read_exact(&mut buffer).unwrap();
+                buffer
+            }
+            None => vec![],
+        }
+    }
+
+    /// Pipe data into [`Child`] stdin in a separate thread to avoid deadlocks. In contrast to
+    /// [`UChild::write_in`], this method is designed to simulate a pipe on the command line and can
+    /// be used only once or else panics. Note, that [`UCommand::set_stdin`] must be used together
+    /// with [`Stdio::piped`] or else this method doesn't work as expected. `Stdio::piped` is the
+    /// current default when using [`UCommand::run_no_wait`]) without calling `set_stdin`. This
+    /// method stores a [`JoinHandle`] of the thread in which the writing to the child processes'
+    /// stdin is running. The associated thread is joined with the main process in the methods below
+    /// when exiting the child process.
+    ///
+    /// * [`UChild::wait`]
+    /// * [`UChild::wait_with_output`]
+    /// * [`UChild::pipe_in_and_wait`]
+    /// * [`UChild::pipe_in_and_wait_with_output`]
+    ///
+    /// Usually, there's no need to join manually but if needed, the [`UChild::join`] method can be
```

Comment:
You've called it `join_in`, though I indeed like just `join` better.

Reply:
oh, I see. Then let's call it `join`.